README.md

```diff
@@ -402,33 +404,76 @@ x-okapi-tenant: [tenant]
 x-okapi-token: [JWT_TOKEN]
 
 {
-  "entityTypes": ['instance', 'subject', 'contributor', 'classification']
+  "entityTypes": ["instance", "subject", "contributor", "classification"]
 }
 ```
 
-The assumption for running this API is that Merge step was run before with ```POST /search/index/instance-records/reindex/full``` which means we have aggregated data in our models.
+We can call the above API as _upload reindex_ API as it runs only Upload step. So, the assumption for running _upload reindex_ is that Merge step was run before with _full reindex_ API.
+If it is not the case or if our changes require to run the Merge step which means there are breaking changes on the mapping between the inventory Instance resources and OpenSearch indices
+then _full reindex_ API.
+
+We can specify one or many resource types in ```entityTypes``` array depending on our needs and if more than one resource type is given then upload step will run for each one of the
+specified resources.
 
 In order to monitor the async processes initiated by the above two APIs we can execute the endpoint:
 ```http
 GET /search/index/instance-records/reindex/status
 ```
 
-and it returns a response that may look like this:
+and it returns a response showing the status details for the latest run operations, merge and upload alike, which may look like the following:
 ```json
-{
-  "entityType": "instance",
-  "status": "Upload Completed",
-  "totalMergeRanges": 3,
-  "processedMergeRanges": 3,
-  "totalUploadRanges": 2,
-  "processedUploadRanges": 2,
-  "startTimeMerge": "2024-04-01T01:37:34.15755006Z",
-  "endTimeMerge": "2024-04-01T01:37:35.15755006Z",
-  "startTimeUpload": "2024-04-01T01:37:36.15755006Z",
-  "endTimeUpload": "2024-04-01T01:37:37.15755006Z"
-}
+[
+  {
+    "entityType": "holding",
+    "status": "Merge Completed",
+    "totalMergeRanges": 5,
+    "processedMergeRanges": 5,
+    "totalUploadRanges": 0,
+    "processedUploadRanges": 0,
+    "startTimeMerge": "2024-04-01T00:38:34.15755006Z",
+    "endTimeMerge": "2024-04-01T00:38:35.15755006Z"
+  },
+  {
+    "entityType": "item",
+    "status": "Merge Completed",
+    "totalMergeRanges": 3,
+    "processedMergeRanges": 3,
+    "totalUploadRanges": 0,
+    "processedUploadRanges": 0,
+    "startTimeMerge": "2024-04-01T00:37:34.15755006Z",
+    "endTimeMerge": "2024-04-01T00:37:35.15755006Z"
+  },
+  {
+    "entityType": "instance",
+    "status": "Upload Completed",
+    "totalMergeRanges": 3,
+    "processedMergeRanges": 3,
+    "totalUploadRanges": 2,
+    "processedUploadRanges": 2,
+    "startTimeMerge": "2024-04-01T00:37:34.15755006Z",
+    "endTimeMerge": "2024-04-01T00:37:35.15755006Z",
+    "startTimeUpload": "2024-04-01T00:37:36.15755006Z",
+    "endTimeUpload": "2024-04-01T00:37:37.15755006Z"
+  },
+  {
+    "entityType": "subject",
+    "status": "Upload Completed",
```

Comment:
The actual statuses use enum names, not values. For example this one will be UPLOAD_COMPLETED. Please fix accordingly.